### PR TITLE
add land region script and update Manifest to branch

### PR DESCRIPTION
**Pull request by kmdeck**
## Purpose 
Create a regional run example which runs when we run long runs. 

In the future, we will move this to a tutorial.

## Content
- long lat swap in Plane constructor
- regional land script

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/shared_utilities/Domains.jl [hunk 1]
```diff
@@ -267,22 +267,27 @@ function Plane(;
         @assert periodic == (false, false)
         radius_earth = FT(radius_earth)
         long, lat = longlat
+        dxlim = xlim # long
+        dylim = ylim # lat
```
Comment (Sbozzolo) [inline src/shared_utilities/Domains.jl:271]:
```suggestion
        dxlim = xlim # lat
        dylim = ylim # long
```
I had to switch things around to work with climacore

File: experiments/long_runs/land_region.jl [hunk 2]
```diff
@@ -0,0 +1,666 @@
+# # Global run of land model
+
+# The code sets up and runs the soil/canopy model for 6 hours on a spherical domain,
```
Comment (Sbozzolo) [inline experiments/long_runs/land_region.jl:3]:
```suggestion
# The code sets up and runs the soil/canopy model for 6 hours on a small region of the globe,
```

File: experiments/long_runs/land_region.jl [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+    τ_NIR_leaf = FT(0.25)
+
+    # Energy Balance model
+    ac_canopy = FT(2.5e4) # this will likely be 10x smaller!
+
+    # Conductance Model
+    g1 = FT(141) # Wang et al: 141 sqrt(Pa) for Medlyn model; Natan used 300.
+
+    #Photosynthesis model
+    Vcmax25 = FT(9e-5) # from Yujie's paper 4.5e-5 , Natan used 9e-5
```
Comment (Sbozzolo) [inline experiments/long_runs/land_region.jl:375]: Maybe we can use the spatially varying VCmax here?

File: experiments/long_runs/land_region.jl [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+    driver_cb = ClimaLand.DriverUpdateCallback(updateat, updatefunc)
+    return prob, SciMLBase.CallbackSet(driver_cb, diag_cb)
+end
+
+function setup_and_solve_problem(; greet = false)
+
+    t0 = 0.0
+    tf = 60 * 60.0 * 24 * 2
+    Δt = 900.0
+    nelements = (2, 2, 15)
```
Comment (Sbozzolo) [inline experiments/long_runs/land_region.jl:628]: There's no reason to run so few elements on a GPU. It is pretty much free to run many more if you want.

File: src/shared_utilities/Domains.jl (hunk 1 above)
Comment (kmdeck) [reply src/shared_utilities/Domains.jl:271]: I thought though that the user passes in xlim refering to long, and ylim refering to lat, and "longlat". And then here internally we swap them to work with climacore. I updated the comments to how I understand it, but let me know if the user actually passes in xlim refering to lat, ylim refering long.

Comment (Sbozzolo) [reply src/shared_utilities/Domains.jl]: Ah yes. You were and are right!

File: experiments/long_runs/land_region.jl [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+        dz_tuple = FT.((10.0, 0.5)),
+    )
+    surface_space = domain.space.surface
+    subsurface_space = domain.space.subsurface
+
+    ref_time = DateTime(2021)
+    t_start = t0
+    # Forcing data
+    era5_artifact_path =
+        ClimaLand.Artifacts.era5_land_forcing_data2021_folder_path(; context)    # Precipitation:
```
Comment (Sbozzolo) [inline experiments/long_runs/land_region.jl:82]:
```suggestion
        ClimaLand.Artifacts.era5_land_forcing_data2021_folder_path(; context)    
        # Precipitation:
```

File: experiments/long_runs/land_region.jl [hunk 6]
```diff
@@ -0,0 +1,676 @@
...
+simdir = ClimaAnalysis.SimDir(outdir)
+short_names = ["gpp", "ct", "swc", "si"]
+for short_name in short_names
+    var = get(simdir; short_name)
+    times = ClimaAnalysis.times(var)
+    for t in times
+        fig = CairoMakie.Figure(size = (800, 600))
+        kwargs = ClimaAnalysis.has_altitude(var) ? Dict(:z => 1) : Dict()
+        tmp = ClimaAnalysis.slice(var, time = t; kwargs...)
+        if ~all(isnan.(tmp.data))
```
Comment (Sbozzolo) [inline experiments/long_runs/land_region.jl:671]:
```suggestion
        if !all(isnan.(tmp.data))
```

`~` is bitwise 'not', `!` is logical 'not'. You are almost never working with bits, so you should pretty much always use `!`

**Review by Sbozzolo — APPROVED** [review]

**Pull request merged**